src/main/java/com/visenze/datatracking/data/Event.java

```diff
@@ -0,0 +1,641 @@
+    private String version;
+
+    @SerializedName("sdk")
+    @Expose
+    private String sdk;
+
+    @SerializedName("pid")
+    @Expose
+    private String pid;
+
+    @SerializedName("imUrl")
+    @Expose
+    private String imageUrl;
+
+    @SerializedName("pos")
+    @Expose
+    private String position;
+
+    @SerializedName("brand")
+    @Expose
+    private String brand;
+
+    @SerializedName("price")
+    @Expose
+    private String price;
+
+    @SerializedName("currency")
+    @Expose
+    private String currency;
+
+    @SerializedName("productUrl")
+    @Expose
+    private String productUrl;
+
+    @SerializedName("transId")
+    @Expose
+    private String transactionId;
+
+    @SerializedName("url")
+    @Expose
+    private String url;
+
+    @SerializedName("r")
+    @Expose
+    private String referrer;
+
+    @SerializedName("p")
+    @Expose
+    private String platform;
+
+    @SerializedName("os")
+    @Expose
+    private String os;
+
+    @SerializedName("osv")
+    @Expose
+    private String osVersion;
+
+    @SerializedName("sr")
+    @Expose
+    private String screenResolution;
+
+    @SerializedName("ab")
+    @Expose
+    private String appId;
+
+    @SerializedName("an")
+    @Expose
+    private String appName;
+
+    @SerializedName("av")
+    @Expose
+    private String appVersion;
+
+    @SerializedName("db")
+    @Expose
+    private String deviceBrand;
+
+    @SerializedName("dm")
+    @Expose
+    private String deviceModel;
+
+    @SerializedName("aaid")
+    @Expose
+    private String aaid;
+
+    @SerializedName("didmd5")
+    @Expose
+    private String didmd5;
+
+    @SerializedName("geo")
+    @Expose
+    private String geo;
+
+    @SerializedName("lang")
+    @Expose
+    private String lang;
+
+    @SerializedName("carrier")
+    @Expose
+    private String carrier;
+
+    @SerializedName("c")
+    @Expose
+    private String campaign;
+
+    @SerializedName("cag")
+    @Expose
+    private String campaignAdGroup;
+
+    @SerializedName("cc")
+    @Expose
+    private String campaignCreative;
+
+    @Expose
+    private String s1;
+
+    @Expose
+    private String s2;
+
+    @Expose
+    private String s3;
+
+    @Expose
+    private String s4;
+
+    @Expose
+    private String s5;
+
+
+
+    protected Event() {
+        Date date = new Date();
+        timestamp = Long.toString(date.getTime());
+    }
+
+    public Map<String, String> toMap() {
+        Gson gson = new Gson();
+        String jStr = gson.toJson(this);
+        return gson.fromJson(jStr, Map.class);
+    }
+
+    public static boolean isValidEvent(Event e) {
+        String action = e.getAction();
+        if(action == null) {
+            return false;
+        }
+        if (action.equals(Constants.Action.SEARCH)) {
+            return (e.queryId != null);
+        } else if (action.equals(Constants.Action.PRODUCT_CLICK) || action.equals(Constants.Action.PRODUCT_VIEW) || action.equals(Constants.Action.ADD_TO_CART)) {
```

Comment:
can remove all the "else". It has the same effect as already return previously.
Make the code easier to read.